fairlearn/metrics/_multi_curve_plot.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+from ..utils._input_validation import (
+    _validate_and_reformat_input,
+    _INCONSISTENT_ARRAY_LENGTH,
+    _INPUT_DATA_FORMAT_ERROR_MESSAGE,
+)
+from ..postprocessing._plotting import _MATPLOTLIB_IMPORT_ERROR_MESSAGE
+from ._make_derived_metric import _DerivedMetric
+from typing import Callable, Union
+from numpy import array
+from sklearn.utils.validation import check_array
+from numpy import zeros
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def plot_model_comparison(
+    *,
+    y_preds,
+    y_true=None,
+    sensitive_features=None,
+    x_axis_metric: Callable[..., Union[float, int]] = None,
+    y_axis_metric: Callable[..., Union[float, int]] = None,
+    ax=None,
+    axis_labels=True,
+    point_labels=False,
+    point_labels_position=(0, 0),
+    legend=False,
+    legend_kwargs={},
+    show_plot=False,
+    **kwargs,
+):
+    r"""
+    Plot a model comparison.
+
+    Parameters
+    ----------
+    y_preds : array-like, dict of array-like
+        An array-like containing predictions per model. Hence, predictions of
+        model :code:`i` should be in :code:`y_preds[i]`.
+
+    y_true : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The ground-truth labels (for classification) or target values (for regression).
+
+    sensitive_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame, optional
+        The sensitive features which should be used to create the subgroups.
+        At least one sensitive feature must be provided.
+        All names (whether on pandas objects or dictionary keys) must be strings.
+        We also forbid DataFrames with column names of ``None``.
+        For cases where no names are provided
+        we generate names ``sensitive_feature_[n]``.
+
+    x_axis_metric : Callable
+        The (aggregating) metric function for the x-axis
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results. For instance, use
+        `make_derived_metric(metric=balanced_accuracy_score, transform='group_min')`
+        to aggregate the `balanced_accuracy_score`.
+
+    y_axis_metric : Callable
+        The (aggregating) metric function for the y-axis, similar to x_axis_metric.
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results.
+
+    ax : matplotlib.axes.Axes, optional
+        If supplied, the scatter plot is drawn on this Axes object.
+        Else, a new figure with Axes is created.
+
+    axis_labels : bool, tuple
```

Comment:
Can we do a list (of length two) instead of a tuple as the explicit option?

Reply:
Yes! But I will also allow tuple then